lib/ansible/modules/cloud/vmware/vmware_host.py

```diff
@@ -283,11 +284,13 @@ def reconnect_host_to_vcenter(self):
         reconnecthost_args['reconnectSpec'].syncState = True
 
         if self.esxi_username is not None or self.esxi_password is not None:
-            reconnecthost_args['cnxSpec'] = self.get_host_connect_spec()
 
             for count in range(0, 2):
                 try:
-                    task = self.host.ReconnectHost_Task(**reconnecthost_args)
+                    if self.folder:
+                        task = self.host.host[0].ReconnectHost_Task(**reconnecthost_args)
+                    elif self.cluster:
+                        task = self.host.ReconnectHost_Task(**reconnecthost_args)
                     success, result = wait_for_task(task)
```

Comment:
IMHO it would be simpler to set self.host as a HostSystem object.
And add a self.host_parent_compute_resource variable for the parent ComputeResource (when self.folder).

the host_parent_compute_resource would only be used for Destroy_Task as: https://www.vmware.com/support/developer/vc-sdk/visdk25pubs/ReferenceGuide/vim.HostSystem.html
> Invoking destroy on a HostSystem of standalone type throws a NotSupported fault. A standalone HostSystem can be destroyed only by invoking destroy on its parent ComputeResource.

Reply:
@ckotte What do you think ?

Reply:
> @ckotte What do you think ?

Yes, I haven't had time for this yet.  Could you provide me some example code?

Reply:
@ckotte https://paste.gnome.org/pck6vwny2

Reply:
Thx. I needed to update the remove method as well.
```
if self.folder:
    task = self.host_parent_compute_resource.Destroy_Task()
elif self.cluster:
    task = self.host.Destroy_Task()
changed, result = wait_for_task(task)
```